File: app/src/main/java/com/mindsilence/game/feature/game/LevelDuration.kt
package com.mindsilence.game.feature.game

fun durationForLevel(level: Int): Int =
    if (level <= 0) 0 else 4 shl (level - 1)

fun totalSessionSeconds(level: Int, elapsedSecAtLevel: Int): Int {
    if (level <= 0) return 0

    var total = elapsedSecAtLevel
    for (currentLevel in 1 until level) {
        total += durationForLevel(currentLevel)
    }
    return total
}

File: Android_maui_app/Domain/UseCases/LevelDuration.cs
namespace MindSilence.Domain.UseCases;

public static class LevelDuration
{
	public static int DurationForLevel(int level) =>
		level >= 1 ? 4 << (level - 1) : 0;

	public static int TotalSessionSeconds(int level, int elapsedSecAtLevel)
	{
		if (level <= 0)
			return 0;

		var total = elapsedSecAtLevel;
		for (var completedLevel = 1; completedLevel < level; completedLevel++)
			total += DurationForLevel(completedLevel);

		return total;
	}
}
